appcues/src/main/java/com/appcues/analytics/Analytics.kt

```diff
@@ -0,0 +1,188 @@
+import com.appcues.analytics.AnalyticsIntent.UpdateProfile
+import com.appcues.analytics.AnalyticsIntentQueue.IntentProcessor
+import com.appcues.data.model.Experience
+import com.appcues.data.model.ExperienceTrigger.Qualification
+import com.appcues.data.model.QualificationResult
+import com.appcues.monitor.ApplicationMonitor
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.launch
+import java.util.Date
+
+internal interface QualificationService {
+
+    // will merge and hit the backend to see if it qualifies, returning QualificationResult
+    suspend fun qualify(intents: List<AnalyticsIntent>): QualificationResult?
+}
+
+internal interface RenderingService {
+
+    // will attempt to show experiences by order
+    // clearCache is mostly used when we know the qualification is from a screen_view
+    suspend fun show(experiences: List<Experience>, clearCache: Boolean)
+
+    suspend fun reset()
+}
+
+internal interface SessionService {
+
+    // returns whether session is started or not
+    suspend fun isSessionStarted(): Boolean
+
+    // starts session (can start using incoming intent or existing stored information
+    suspend fun startSession(intent: AnalyticsIntent): Boolean
+
+    // decorates incoming intent and returns a new intent with session/auto props information
+    suspend fun decorateIntent(intent: AnalyticsIntent): AnalyticsIntent
+
+    suspend fun reset()
+}
+
+// new model for Intents available for Analytics
+internal sealed class AnalyticsIntent(open val isInternal: Boolean, val timestamp: Date = Date()) {
+
+    object Anonymous : AnalyticsIntent(false)
+
+    data class Identify(
+        val userId: String,
+        val properties: Map<String, Any>?
+    ) : AnalyticsIntent(false)
+
+    data class UpdateProfile(
+        val properties: Map<String, Any>?,
+        override val isInternal: Boolean
+    ) : AnalyticsIntent(isInternal)
+
+    data class UpdateGroup(
+        val groupId: String?,
+        val properties: Map<String, Any>?,
+        override val isInternal: Boolean
+    ) : AnalyticsIntent(isInternal)
+
+    data class Event(
+        val name: String,
+        val attributes: Map<String, Any>?,
+        val context: Map<String, Any>?,
+        override val isInternal: Boolean
+    ) : AnalyticsIntent(isInternal)
+}
+
+internal class Analytics(
+    private val appcuesCoroutineScope: AppcuesCoroutineScope,
+    private val intentQueue: AnalyticsIntentQueue,
+    private val qualificationService: QualificationService,
+    private val renderingService: RenderingService,
+    private val sessionService: SessionService,
+) : IntentProcessor, ApplicationMonitor.Listener {
+
+    companion object {
+
+        const val EVENT_ATTR_SCREEN_TITLE = "screenTitle"
+        const val EVENT_CONTEXT_SCREEN_TITLE = "screen_title"
+    }
+
+    init {
+        intentQueue.setProcessor(this)
+
+        ApplicationMonitor.subscribe(this)
```

Comment:
ApplicationMonitor here is simplifying an existing logic on SessionMonitor that clears analytics when we know the application lifecycle calls onStopped()